contracts/upgradeable_contracts/erc20_to_native/HomeBridgeErcToNative.sol

```diff
@@ -47,24 +47,53 @@ contract HomeBridgeErcToNative is EternalStorage, BasicBridge, BasicHomeBridge,
         address _owner
     ) public returns(bool)
     {
-        require(!isInitialized());
-        require(_validatorContract != address(0) && isContract(_validatorContract));
-        require(_requiredBlockConfirmations > 0);
-        require(_minPerTx > 0 && _maxPerTx > _minPerTx && _dailyLimit > _maxPerTx);
-        require(_blockReward == address(0) || isContract(_blockReward));
-        require(_foreignMaxPerTx < _foreignDailyLimit);
-        require(_owner != address(0));
-        addressStorage[keccak256(abi.encodePacked("validatorContract"))] = _validatorContract;
-        uintStorage[keccak256(abi.encodePacked("deployedAtBlock"))] = block.number;
-        uintStorage[keccak256(abi.encodePacked("dailyLimit"))] = _dailyLimit;
-        uintStorage[keccak256(abi.encodePacked("maxPerTx"))] = _maxPerTx;
-        uintStorage[keccak256(abi.encodePacked("minPerTx"))] = _minPerTx;
-        uintStorage[keccak256(abi.encodePacked("gasPrice"))] = _homeGasPrice;
-        uintStorage[keccak256(abi.encodePacked("requiredBlockConfirmations"))] = _requiredBlockConfirmations;
-        addressStorage[keccak256(abi.encodePacked("blockRewardContract"))] = _blockReward;
-        uintStorage[keccak256(abi.encodePacked("executionDailyLimit"))] = _foreignDailyLimit;
-        uintStorage[keccak256(abi.encodePacked("executionMaxPerTx"))] = _foreignMaxPerTx;
-        setOwner(_owner);
+        _initialize(
+            _validatorContract,
+            _dailyLimit,
+            _maxPerTx,
+            _minPerTx,
+            _homeGasPrice,
+            _requiredBlockConfirmations,
+            _blockReward,
+            _foreignDailyLimit,
+            _foreignMaxPerTx,
+            _owner
+        );
+        setInitialize(true);
+
+        return isInitialized();
+    }
+
+    function rewardableInitialize (
+        address _validatorContract,
+        uint256 _dailyLimit,
+        uint256 _maxPerTx,
+        uint256 _minPerTx,
+        uint256 _homeGasPrice,
+        uint256 _requiredBlockConfirmations,
+        address _blockReward,
+        uint256 _foreignDailyLimit,
+        uint256 _foreignMaxPerTx,
+        address _owner,
+        address _feeManager,
+        uint256 _fee
+    ) public returns(bool)
+    {
+        _initialize(
+            _validatorContract,
+            _dailyLimit,
+            _maxPerTx,
+            _minPerTx,
+            _homeGasPrice,
+            _requiredBlockConfirmations,
+            _blockReward,
+            _foreignDailyLimit,
+            _foreignMaxPerTx,
+            _owner
+        );
+        require(isContract(_feeManager));
+        addressStorage[keccak256(abi.encodePacked("feeManagerContract"))] = _feeManager;
+        require(_feeManager.delegatecall(abi.encodeWithSignature("setFee(uint256)", _fee)));
```

Comment:
what if we make an internal method `_setFee` in `RewardableBridge.sol`:
```solidity
function _setFee(address _feeManager, uint256 _fee) internal {
    require(_feeManager.delegatecall(abi.encodeWithSignature("setFee(uint256)", _fee)));
}
```
This method will be called by `setFee` of the `RewardableBridge` contract and by `rewardableInitialize` of the `HomeBridgeErcToNative` contract:
```solidity
require(_setFee(_feeManager, _fee));
```
It will keep the logic of setting the parameter in one place.

Reply:
Thanks for suggestion. Done https://github.com/poanetwork/poa-bridge-contracts/pull/134/commits/0576f184e2820c47be346aa6e18c2ee0bcc2931c